mne/channels/channels.py

```diff
@@ -440,18 +440,36 @@ def inst_has(attr):
 class InterpolationMixin(object):
     """Mixin class for Raw, Evoked, Epochs
     """
-    def interpolate_bads_eeg(self):
-        """Interpolate bad channels
 
-        Operates in place.
+    def interpolate_bads(self, reset_bads=True):
+        """Interpolate bad MEG and EEG channels.
+
+        Parameters
+        ----------
+        reset_bads : bool
+            If True, remove the bads from info.
+
+        Returns
+        -------
+        inst : mne.io.Raw, mne.Epochs or mne.Evoked
+            The interpolated data.
 
         Notes
         -----
         .. versionadded:: 0.9.0
         """
         from .interpolation import _interpolate_bads_eeg
-        _interpolate_bads_eeg(self)
-        return self
+        from ..forward import _interpolate_bads_meg
+
+        if getattr(self, 'preload', None) is False:
+            raise ValueError('Data must be preloaded.')
+
+        inst = _interpolate_bads_meg(_interpolate_bads_eeg(self.copy()))
```

Comment:
I asked because we make a copy with virtual channels but I guess this is a different situation and we update only couple of channels compared to all channels.